framework/src/binding/state/trie_db.rs

```diff
@@ -26,8 +34,21 @@ impl RocksTrieDB {
         Ok(RocksTrieDB {
             light,
             db: Arc::new(db),
+            cache: RwLock::new(HashMap::new()),
         })
     }
+
+    fn inner_get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, RocksTrieDBError> {
+        let res = {
+            let cache = self.cache.read();
+            cache.get(key).cloned()
+        };
+
+        if res.is_none() {
+            return Ok(self.db.get(key).map_err(to_store_err)?);
```

Comment:
If not exists in that cache, should insert it after getting from the DB?  
Another question, what if is full of that cache at this moment?

Reply:
If you insert the obtained data into the cache here, it is no different from LRU. Doing so cannot achieve true read concurrency.